src/parsers/utils/fortran.js

```diff
@@ -0,0 +1,127 @@
+import { regex } from "./settings";
+
+const typeParsers = {
+    [Number]: (value) => parseFloat(value),
+    [String]: (value) => value,
+    [Boolean]: (value) => value === "true",
+};
+
+export const FortranParserMixin = (superclass) =>
+    class extends superclass {
+        /**
+         * Parses Fortran namelists and cards data from a string.
+         *
+         * @summary Parses Fortran namelists and cards data from a QE input file string.
+         * @param {String} content - The text to parse.
+         * @throws {Error} If no namelist data is found in `text`.
+         * @throws {Error} If no cards data is found in `text`.
+         * @returns {Object} An object containing the parsed namelist and cards data. The exact structure of this object will depend on the structure of the namelist and cards data in `text`.
+         */
+        parseNamelists(content) {
```

Comment:
Let's wrap up the functions in `fortran...` to make it easier to find where they came from?

fortranParseNamelist
fortranExtractPairs
etc.